File: BitBoy/app/src/main/java/com/ironraft/bitboy/LanguageFactory.java
package com.ironraft.bitboy;


import android.app.Activity;
import android.content.SharedPreferences;
import android.graphics.Typeface;
import android.util.Log;

import com.ironraft.bitboy.model.AppManager;
import com.ironraft.bitboy.model.TopicManager;


public class LanguageFactory {


    private static LanguageFactory instence;
    private final String LANG_KEY="LANG_KEY";

    public final static String ENG = "eng";
    public final static String KOR = "kor";
    private String type = ENG;
    private SharedPreferences shared;

	public static LanguageFactory getInstence()
    {
        if(instence == null) instence = new LanguageFactory();
        return instence;
	}

    public LanguageFactory()
    {
    	instence=this;
        shared = MainActivity.getInstence().getSharedPreferences(Config.PREFS_NAME, 0);
        type = getLanguageType();
    }

    public String getLanguageType()
    {
        return shared.getString(LANG_KEY,ENG);
    }
    public void setLanguageType(String _type)
    {
        TopicManager.getInstence().unRegisterLanguage();
        type = _type;
        SharedPreferences.Editor editor = shared.edit();
        editor.putString(LANG_KEY,type);
        editor.commit();

        MainActivity.getInstence().update();
        AppManager.getInstence().updatePushToken();
        TopicManager.getInstence().registerLanguage();
    }

    public int getResorceID(String value)
    {
        int rid = MainActivity.getInstence().getResources().getIdentifier( value+"_"+type, "string", MainActivity.getInstence().getPackageName());
        return rid;
    }
    public String getResorceString(String value)
    {

        int rid = getResorceID(value);
        String str = MainActivity.getInstence().getResources().getString(rid);
        return str;

    }




}

File: SMSManager/app/src/main/java/lib/core/LanguageFactory.kt
package lib.core

import android.content.SharedPreferences

class LanguageFactory
{
    private val LANG_KEY = "LANG_KEY"
    private var type = KOR
    private val shared: SharedPreferences?

    var languageType: String
        get() = shared!!.getString(LANG_KEY, KOR)
        set(_type)
        {
            type = _type
            val editor = shared!!.edit()
            editor.putString(LANG_KEY, type)
            editor.commit()
        }

    init
    {
        instence = this
        shared = ActivityCore.shareInstence()?.getSharedPreferences(ActivityCore.PREFS_NAME, 0)
        type = languageType
    }

    fun getResorceID(value: String): Int
    {
        if(ActivityCore.shareInstence() == null) return -1
        return ActivityCore.shareInstence()!!.getResources().getIdentifier(value + "_" + type, "string",
                ActivityCore.shareInstence()!!.getPackageName())
    }

    fun getResorceString(value: String): String
    {
        if(ActivityCore.shareInstence() == null) return ""
        val rid = getResorceID(value)
        return ActivityCore.shareInstence()!!.getResources().getString(rid)

    }

    companion object
    {

        private var instence: LanguageFactory? = null
        val ENG = "eng"
        val KOR = "kor"

        fun shareInstence(): LanguageFactory
        {
            if (instence == null) instence = LanguageFactory()
            return instence!!
        }
    }


}